src/System.Private.ServiceModel/tests/Scenarios/Extensibility/WebSockets/WebSocketTests.cs

```diff
@@ -598,4 +632,456 @@ public static void WebSocket_WSScheme_WSTransportUsageAlways_DuplexCallback_Guid
             ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)proxy, factory);
         }
     }
+
+    [Fact]
+    [OuterLoop]
+    public static void WebSocket_Https_RequestReply_BinaryBuffered()
+    {
+        BinaryMessageEncodingBindingElement binaryMessageEncodingBindingElement = null;
+        HttpsTransportBindingElement httpsTransportBindingElement = null;
+        CustomBinding binding = null;
+        ChannelFactory<IWSRequestReplyService> channelFactory = null;
+        IWSRequestReplyService client = null;
+
+        try
+        {
+            // *** SETUP *** \\
+            binaryMessageEncodingBindingElement = new BinaryMessageEncodingBindingElement();
+            httpsTransportBindingElement = new HttpsTransportBindingElement()
+            {
+                MaxReceivedMessageSize = ScenarioTestHelpers.DefaultMaxReceivedMessageSize,
+                MaxBufferSize = ScenarioTestHelpers.DefaultMaxReceivedMessageSize,
+            };
+            httpsTransportBindingElement.WebSocketSettings.TransportUsage = WebSocketTransportUsage.Always;
+            binding = new CustomBinding(binaryMessageEncodingBindingElement, httpsTransportBindingElement);
+
+            channelFactory = new ChannelFactory<IWSRequestReplyService>(binding, new EndpointAddress(Endpoints.WebSocketHttpsRequestReplyBinaryBuffered_Address));
+            client = channelFactory.CreateChannel();
+
+            // *** EXECUTE *** \\
+            // Invoking DownloadData
+            string result = client.DownloadData();
+
+            // Invoking UploadData
+            client.UploadData(ScenarioTestHelpers.CreateInterestingString(123));
+
+            // *** VALIDATE *** \\
+            foreach (string serverLogItem in client.GetLog())
+            {
+                Assert.True(serverLogItem != ScenarioTestHelpers.RemoteEndpointMessagePropertyFailure, ScenarioTestHelpers.RemoteEndpointMessagePropertyFailure);
```

Comment:
Ah, I see the message is: "RemoteEndpointMessageProperty did not contain the address of this machine". I think that should be descriptive enough, though "this" is ambiguous in the context of a x-machine test